velox/functions/prestosql/SIMDJsonFunctions.h

```diff
@@ -118,10 +118,14 @@ struct SIMDJsonArrayContainsFunction {
             }
           }
         }
+      } catch (const simdjson::simdjson_error& e) {
+        if (e.error() != simdjson::INCORRECT_TYPE &&
```

Comment:
When do we get INCORRECT_TYPE and NUMBER_ERROR? Which of these errors do we get in the new test cases? Can we make sure to add tests for both errors? It would be nice to add a comment here to provide examples of each of these errors and explain why we allow these.

Reply:
In `SIMDJsonArrayContainsFunction`, we first judge `v.type()`, and then get the value of the specified type. And v.type() is as shown below. The type is not judged based on the complete json value and it may be wrong.
```
simdjson_inline simdjson_result<json_type> value_iterator::type() const noexcept {
  switch (*peek_start()) {
    case '{':
      return json_type::object;
    case '[':
      return json_type::array;
    case '"':
      return json_type::string;
    case 'n':
      return json_type::null;
    case 't': case 'f':
      return json_type::boolean;
    case '-':
    case '0': case '1': case '2': case '3': case '4':
    case '5': case '6': case '7': case '8': case '9':
      return json_type::number;
    default:
      return TAPE_ERROR;
  }
}
```

Next, we will discuss the types of errors that occur in different implementations for the input `value` type. 

1. `value` is bool:
For `get_bool()`, if the corresponding json is not "true" or "false", the error code is "INCORRECT_TYPE".

2. `value` is int64_t:
For `get_int64()`, if the json value is within the range of int64, it is considered legal. If it is all numbers, but not within this range or has 'e' or 'E' or '.', the error code is INCORRECT_TYPE. Otherwise, for example, if there are illegal characters,  the error code is NUMBER_ERROR.

3. `value` is double:
For `get_double()`, if the json value is 0, the error code is INCORRECT_TYPE, and other exceptions are NUMBER_ERROR.

4. `value` is string:
For `get_string()`, if the json value does not start with ", the error code is INCORRECT_TYPE, and other exceptions are STRING_ERROR.

If there are multiple json values in the json array, and some of them are illegal, I think it should not return null directly, and it should continue to judge whether there is a json value that matches the input `value`. For example, 
```
jsonArrayContains<double>("[9.6E400]", 4.2) => false;
jsonArrayContains<double>("[9.6E400, 4.2]", 4.2) => true;
```

Reply:
I've added some comments for clarification, and I've added as many test cases as possible to cover these types of errors.

It should be noted that in the implementation, the type is judged first and then the json value of the specified type is obtained, so some errors will not be triggered. For example, for the double type, `get_number_type()` has intercepted illegal cases, so `get_double()` will not trigger "INCORRECT_TYPE" error. All cases that trigger different error types have been added to the test file.

Reply:
Thank you for all these detailed explanations. Appreciate more test coverage.